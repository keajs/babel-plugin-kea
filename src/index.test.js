/* global test, describe, expect */
import { transform } from '@babel/core'
import keaPlugin from './index'

describe('babel-plugin-kea', () => {
  test('works with kea 2.0 logic input', () => {
    const oldCode = `
      const logic = kea({
        actions: { doit: true, }
      })
    `

    const { code } = transform(oldCode, {
      envName: 'production',
      code: true,
      babelrc: false,
      configFile: false,
      filename: 'scenes/page/index.ts',
      plugins: [keaPlugin],
    })

    expect(code).toMatchSnapshot()
  })

  test('works with kea 3.0 logic builders', () => {
    const oldCode = `
      const logic = kea([
        actions({ doit: true }),
      ])
    `

    const { code } = transform(oldCode, {
      envName: 'production',
      code: true,
      babelrc: false,
      configFile: false,
      filename: 'scenes/page/index.ts',
      plugins: [keaPlugin],
    })

    expect(code).toMatchSnapshot()
  })
})
