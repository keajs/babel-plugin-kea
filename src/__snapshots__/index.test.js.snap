// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-kea works with kea 2.0 logic input 1`] = `
"const logic = kea({
  path: [\\"scenes\\", \\"page\\", \\"index\\"],
  actions: {
    doit: true
  }
});"
`;

exports[`babel-plugin-kea works with kea 3.0 logic builders 1`] = `
"import { kea, path } from 'kea';
const logic = kea([path([\\"scenes\\", \\"page\\", \\"index\\"]), actions({
  doit: true
})]);"
`;

exports[`babel-plugin-kea works with kea 3.0 logic builders, no import 1`] = `
"const logic = kea([path([\\"scenes\\", \\"page\\", \\"index\\"]), actions({
  doit: true
})]);"
`;
